File: web/src/render/commentRenderer.ts
import type MarkdownNode from "../types/MarkdownNode";
import type Renderer from "../types/Renderer";
import type RendererState from "../types/RendererState";
import renderChildren from "./renderChildren";
import { endNewLine, startNewLine } from "./renderUtils";

const renderer: Renderer = {
	name: "comment",
	render,
};
export default renderer;

function render(node: MarkdownNode, state: RendererState): void {
	startNewLine(node, state);
	state.output += `<span class="markdown-comment">`;
	renderChildren(node, state);
	state.output += "</span>";
	endNewLine(node, state);
}

File: zig/src/render/commentRenderer.zig
const std = @import("std");
const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const RendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;
const renderTag = @import("renderTag.zig").renderTag;

const renderChildren = @import("renderChildren.zig").renderChildren;
const renderUtils = @import("renderUtils.zig");

pub fn render(node: *const MarkdownNode, state: *RendererState, decode: ?bool) void {
    _ = decode;
    renderUtils.startNewLine(node, state);
    state.output.appendSlice(state.allocator, "<span class=\"markdown-comment\">") catch unreachable;
    renderChildren(node, state, true);
    state.output.appendSlice(state.allocator, "</span>") catch unreachable;
    renderUtils.endNewLine(node, state);
}

pub const commentRenderer = Renderer{
    .name = "comment",
    .render = render,
};
